### DISCUSSION TITLE
Lua script does not alter record

### DISCUSSION THREAD

**Discussion by dominikhei** [fluent/fluent-bit#8616] [Q&A]
Hello, I have the following Filter, that should add a few keys to my json record:

```
    [FILTER]
        Name Lua
        Match *
        call add_datetime_fields
        code function add_datetime_fields(tag, timestamp, record) new_record = record new_record["year"] = record["date"]:sub(1, 4) new_record["month"] = record["date"]:sub(6, 7) new_record["day"] = record["date"]:sub(9, 10) new_record["time"] = record["date"]:sub(12, 13) return 2, timestamp, new_record end
```
However the logs that are output by fluentbit still look like this:

`{"date":"2024-03-21T11:23:04.671154Z","log": "..."}`

The new keys are not present, does anybody see what I did wrong?

**Comment by patrick-stephens** [comment]
It's hard to debug a single line LUA filter, does it work with https://fluent.github.io/lua-sandbox/?

**Reply by dominikhei** [reply]
No it does not work, this is the lua function across multiple lines:

```
function add_datetime_fields(tag, timestamp, record) 
  new_record = record 
  new_record["year"] = record["date"]:sub(1, 4) 
  new_record["month"] = record["date"]:sub(6, 7) 
  new_record["day"] = record["date"]:sub(9, 10) 
  new_record["time"] = record["date"]:sub(12, 13) 
  return 1, timestamp, new_record 
end
```

**Reply by patrick-stephens** [reply]
Sounds like your LUA code is problematic then.
Make sure to update the sandbox input data and call your function `cb_filter`.
I would split out the usage of string extraction into separate local variables as well.

**Reply by dominikhei** [reply]
If I rename the function accordingly, everything works in the lua sandbox

**Reply by patrick-stephens** [reply]
OK, well that runs Fluent Bit underneath so sounds like it inherently works. My guess then is something else is wrong with your full config.

**Reply by dominikhei** [reply]
Alright @patrick-stephens then I will try to change some configs. Thanks for pointing out the Lua-sandbox, that is helpful for testing.

Quick question, can I somehow use the tag from the input inside a Lua filter?

**Reply by patrick-stephens** [reply]
Sure, it's part of the function signature...
